src/mbgl/programs/program.hpp

```diff
@@ -58,16 +58,19 @@ class Program {
               const SegmentVector<Attributes>& segments,
               const PaintPropertyBinders& paintPropertyBinders,
               const typename PaintProperties::PossiblyEvaluated& currentProperties,
-              float currentZoom) {
+              float currentZoom,
+              const std::string& layerID) {
         typename AllUniforms::Values allUniformValues = uniformValues
             .concat(paintPropertyBinders.uniformValues(currentZoom, currentProperties));
 
         typename Attributes::Bindings allAttributeBindings = LayoutAttributes::bindings(layoutVertexBuffer)
             .concat(paintPropertyBinders.attributeBindings(currentProperties));
 
         for (auto& segment : segments) {
-            if (!segment.vertexArray) {
-                segment.vertexArray = context.createVertexArray();
+            optional<gl::VertexArray>& vertexArray = segment.vertexArrays[layerID];
+
+            if (!vertexArray) {
+                vertexArray = context.createVertexArray();
```

Comment:
Instead of using `optional` here, can we use key presence as an indicator for whether we need to create the vertex array?